### PR TITLE
add a couple of tests for users screen

### PR DESCRIPTION
**Pull request by maskin25**
# What this PR does

There are the following tests added:

 - admin is allowed to edit other profiles
 - editor is not allowed to edit other profiles

## Which issue(s) this PR fixes

https://github.com/grafana/oncall/issues/1586

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: grafana-plugin/integration-tests/users/usersActions.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { test, expect, Page } from '../fixtures';
+import { goToOnCallPage } from '../utils/navigation';
+
+test.describe('view list of users', () => {
+  test.slow(); // this test is doing a good amount of work, give it time
```
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:5]:
```suggestion
```

File: grafana-plugin/integration-tests/users/usersActions.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { test, expect, Page } from '../fixtures';
+import { goToOnCallPage } from '../utils/navigation';
+
+test.describe('view list of users', () => {
```
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:4]:
```suggestion
test.describe("edit other user's profiles", () => {
```

File: grafana-plugin/integration-tests/users/usersActions.test.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  test.slow(); // this test is doing a good amount of work, give it time
+
+  test('admin is allowed to edit other profiles', async ({ adminRolePage }) => {
```
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:7]:
```suggestion
  test('admin is allowed', async ({ adminRolePage }) => {
```

File: grafana-plugin/integration-tests/users/usersActions.test.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    const usersTableElement = page.getByTestId('users-table');
+    await usersTableElement.waitFor({ state: 'visible' });
+
+    const editOtherUserProfileDisabledButtonList = await page.locator('button.edit-other-profile-button[disabled]');
+
+    expect(editOtherUserProfileDisabledButtonList).toHaveCount(0);
+  });
+
+  test('editor is not allowed to edit other profiles', async ({ editorRolePage }) => {
```
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:20]:
```suggestion
  test('editor is not allowed', async ({ editorRolePage }) => {
```

File: grafana-plugin/integration-tests/users/usersActions.test.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const { page } = adminRolePage;
+
+    await goToOnCallPage(page, 'users');
+
+    const usersTableElement = page.getByTestId('users-table');
+    await usersTableElement.waitFor({ state: 'visible' });
+
+    const editOtherUserProfileDisabledButtonList = await page.locator('button.edit-other-profile-button[disabled]');
+
+    expect(editOtherUserProfileDisabledButtonList).toHaveCount(0);
```
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:17]: I would deduplicate all of this into a function just above, and the only argument to the function would be the `Locator` object that differ between the admin/editor tests, this way you can reuse this logic

File: grafana-plugin/integration-tests/users/usersActions.test.ts
Comment (joeyorlando) [inline grafana-plugin/integration-tests/users/usersActions.test.ts:1]: ❤️

**Review by joeyorlando — APPROVED** [review]
good work! There's two lines you'll also need to update in `oncall-private` [here](https://github.com/search?q=repo%3Agrafana%2Foncall-private%20yarn%20test&type=code) regarding the `yarn test:integration` -> `yarn test:e2e` change

**Comment by teodosii** [comment]
- Admin is allowed to edit other users' profile
- Viewer is not allowed to view the list of users
- Viewer cannot access restricted tabs from View My Profile
- Editor cannot view other users' data
- Editor can access tabs from View My Profile
- Editor is not allowed to edit other users' profile

So far this is the list for e2e permissions on the Users screen. There's room for some other scenarios, worth digging further into it at a later point.

**Review by joeyorlando — COMMENTED** [review]
anything blocking this PR from being merged?

**Review by teodosii — APPROVED** [review]

**Review by joeyorlando — APPROVED** [review]
nice! LGTM

**Pull request merged**